api/controllers/networks.js

```diff
@@ -0,0 +1,258 @@
+  }
+  if (networkQuery.cpvs) {
+    filters.push(`{ as: bids }.out('AppliedTo').in('Comprises').out('HasCPV')
+      { class: CPV, where: (code in :cpvs)}`);
+  }
+  return filters;
+}
+
+function settingsToValueQuery(sizeSetting) {
+  let value;
+  if (sizeSetting === 'numberOfWinningBids') {
+    value = 'count(bids)';
+  } else if (sizeSetting === 'amountOfMoneyExchanged') {
+    value = 'sum(bids.price.netAmountEur)';
+  }
+  return value;
+}
+
+function createBidderNodes(bidFilters, valueQuery, network) {
+  const bidderNodesQuery = `SELECT bidder.name as label,
+    bidder[@rid] as \`@rid\`,
+    ${valueQuery} as value,
+    median(bids.out('AppliedTo').bidsCount) as medianCompetition
+    FROM (
+      MATCH { class: Bidder, as: bidder },
+        { as: bidder }.out('Participates'){ class: Bid, as: bids },
+        ${_.join(bidFilters, ',')}
+      RETURN bidder, bids
+    ) GROUP BY bidder;`;
+  return config.db.query(bidderNodesQuery, { params: network.query })
+    .then((bidderNodes) => Promise.map(bidderNodes, (bidderNode) => {
+      const nodeAttrs = _.pick(
+        bidderNode,
+        ['label', 'value', 'medianCompetition'],
+      );
+      nodeAttrs.type = 'bidder';
+      nodeAttrs.visible = true;
+      nodeAttrs.id = uuidv4();
+      return createActorNode(nodeAttrs, bidderNode['@rid'], network['@rid']);
+    }));
+}
+
+function createBuyerNodes(bidFilters, valueQuery, network) {
+  const buyerNodesQuery = `SELECT buyer.name as label,
+    buyer[@rid] as \`@rid\`,
+    ${valueQuery} as value,
+    buyer.address.country as country,
+    median(bids.out('AppliedTo').bidsCount) as medianCompetition
+    FROM (
+      MATCH { class: Buyer, as: buyer },
+        { as: buyer }.out('Awards'){ class: Bid, as: bids },
+        ${_.join(bidFilters, ',')}
+      RETURN buyer, bids
+    ) GROUP BY buyer;`;
+  return config.db.query(buyerNodesQuery, { params: network.query })
+    .then((buyerNodes) => Promise.map(buyerNodes, (buyerNode) => {
+      const nodeAttrs = _.pick(
+        buyerNode,
+        ['label', 'id', 'value', 'medianCompetition', 'country'],
+      );
+      nodeAttrs.type = 'buyer';
+      nodeAttrs.visible = true;
+      nodeAttrs.id = uuidv4();
+      return createActorNode(nodeAttrs, buyerNode['@rid'], network['@rid']);
+    }));
+}
+
+function createActorNode(nodeAttrs, actorRID, networkRID) {
+  let node;
+  return config.db.create('vertex', 'NetworkActor')
+    .set(nodeAttrs)
+    .commit()
+    .one()
+    .then((createdNode) => {
+      node = createdNode;
+      const networkEdge = config.db.create('edge', 'PartOf')
+        .from(createdNode['@rid'])
+        .to(networkRID)
+        .commit()
+        .one();
+      const actorEdge = config.db.create('edge', 'ActingAs')
+        .from(actorRID)
+        .to(createdNode['@rid'])
+        .commit()
+        .one();
+      return Promise.all([networkEdge, actorEdge]);
+    })
+    .then(() => node);
```

Comment:
@nightsh I don't like this either but I don't know how to create the edges but still return the node. Any ideas?